app/services/credentials_verifier.go

```diff
@@ -45,5 +47,16 @@ func CredentialsVerifier(store data.AccountStore, cfg *app.Config, username stri
 		return nil, FieldErrors{{"credentials", ErrExpired}}
 	}
 
+	//Check TOTP MFA
+	if account.TOTPEnabled() {
+		secret, err := compat.Decrypt([]byte(account.TOTPSecret.String), cfg.DBEncryptionKey)
+		if err != nil {
+			return nil, errors.Wrap(err, "TOTPDecrypt")
+		}
+		if !totp.Validate(totpCode, secret) {
+			return nil, FieldErrors{{"totp", ErrFailed}}
+		}
+	}
```

Comment:
I have this block in a few different places, should I make a function somewhere in lib or maybe a service that takes an account and an encryption key and checks if it's valid?